### PR TITLE
Add logic to resume ImageNet example with new learning rate

### LINKED ISSUE #816: Resume will override learning rate on command line.
If you resume from a checkpoint and want to run with a new learning rate, the saved learning rate will overwrite what you specify on the command line. The fix is simple.

### PR DESCRIPTION
**Pull request by kiersten-stokes**
Fixes #816 

Resume no longer overwrites learning rate specific on command line. 

(My linter also removed 2 extraneous spaces in a `dict`, I can revert this if desired 🙂)

### CODE REVIEW

File: imagenet/main.py [hunk 1]
```diff
@@ -215,6 +215,9 @@ def main_worker(gpu, ngpus_per_node, args):
                 best_acc1 = best_acc1.to(args.gpu)
             model.load_state_dict(checkpoint['state_dict'])
             optimizer.load_state_dict(checkpoint['optimizer'])
+            if args.lr:
```
Comment (soumith) [inline imagenet/main.py:218]: wouldn't `if args.lr` always evaluate to true?

Comment (kiersten-stokes) [reply imagenet/main.py]: shoot, yes, thank you for pointing that out! I must have been looking at `start_epoch` or something

**Comment by kiersten-stokes** [comment]
@soumith just pinging again to see if this is in a good state for approval 🙂
